lib/services/local/externalauditstorage.go

```diff
@@ -109,22 +113,28 @@ func (s *ExternalAuditStorageService) UpsertDraftExternalAuditStorage(ctx contex
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	item := backend.Item{
+		Key:   draftExternalAuditStorageBackendKey,
+		Value: value,
+	}
 
-	// Lock is used here and in Promote to prevent upserting in the middle
-	// of a promotion and the possibility of deleting a newly upserted draft
-	// after the previous one was promoted.
-	err = backend.RunWhileLocked(ctx, backend.RunWhileLockedConfig{
-		LockConfiguration: backend.LockConfiguration{
-			Backend:  s.backend,
-			LockName: externalAuditStorageLockName,
-			TTL:      externalAuditStorageLockTTL,
+	integrationKey, integrationRevision, err := s.checkAWSIntegration(ctx, in.Spec.IntegrationName)
+	if err != nil {
+		return nil, trace.Wrap(err, "checking AWS OIDC integration")
+	}
+	_, err = s.backend.AtomicWrite(ctx, []backend.ConditionalAction{
+		{
+			// Make sure the AWS OIDC integration checked above hasn't changed.
+			Key:       integrationKey,
+			Condition: backend.Revision(integrationRevision),
+			Action:    backend.Nop(),
+		},
+		{
+			// Upsert the new draft EAS integration.
+			Key:       item.Key,
+			Condition: backend.Whatever(),
+			Action:    backend.Put(item),
 		},
-	}, func(ctx context.Context) error {
-		_, err = s.backend.Put(ctx, backend.Item{
-			Key:   draftExternalAuditStorageBackendKey,
-			Value: value,
-		})
-		return trace.Wrap(err)
 	})
 	return in, trace.Wrap(err)
```

Comment:
The returned `*externalauditstorage.ExternalAuditStorage` should have the revision returned by the successful `AtomicWrite`; not sure if it's necessary to clone it or not, but considering that this is a configuration change that happens rarely (and thus isn't very perf-critical) it might be worth doing.